### PR TITLE
Remove FlatBuffers dependency and support

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: tests/connection/broker.cpp [hunk 1]
```diff
@@ -33,12 +33,10 @@ class Subscriber
...
     {
         zeroeq::Subscriber subscriber( test::buildURI( "127.0.0.1", *_publisher ));
-        ::test::SerializablePtr echoEvent = ::test::getFBEchoInEvent(
-                    [this]( const zeroeq::FBEvent& event )
-                    { test::onEchoEvent( event );
-                      received = true; });
 
-        BOOST_CHECK( subscriber.subscribe( *echoEvent ));
+        BOOST_CHECK( subscriber.subscribe( test::Echo::IDENTIFIER(),
+             zeroeq::EventPayloadFunc([&]( const void* data, const size_t size )
```
Comment (hernando) [inline tests/connection/broker.cpp]: You can pass the lambda directly as the argument without invoking the zeroeq::EventPayloadFunc constructor.

File: tests/receiver.cpp [hunk 2]
```diff
@@ -82,28 +71,17 @@ BOOST_AUTO_TEST_CASE(test_publisher_routing)
...
-                    std::bind( &onEvent1, std::placeholders::_1 ));
-
-    BOOST_CHECK( subscriber1->subscribe( *echoEvent1 ));
-
-    ::test::SerializablePtr echoEvent2 = ::test::getFBEchoInEvent(
-                    std::bind( &onEvent2, std::placeholders::_1 ));
-
-    BOOST_CHECK( subscriber2.subscribe( *echoEvent2 ));
+    BOOST_CHECK( subscriber1->subscribe( test::Echo::IDENTIFIER(), &onEvent1 ));
+    BOOST_CHECK( subscriber2.subscribe( test::Echo::IDENTIFIER(), &onEvent2 ));
```
Comment (hernando) [inline tests/receiver.cpp]: Use lambdas instead of free functions and global variables.

File: tests/pubSub.cpp [hunk 3]
```diff
@@ -229,40 +188,35 @@ BOOST_AUTO_TEST_CASE(publish_receive_filters)
...
     }
     const auto& noEchoTime = std::chrono::high_resolution_clock::now() -
                              startTime;
 
     // Benchmark with echo handler, now should send data
-    event = ::test::getFBEchoOutEvent( message );
-    ::test::SerializablePtr largeEchoEvent = ::test::getFBEchoInEvent(
-                    std::bind( &onLargeEcho, std::placeholders::_1 ));
-    BOOST_CHECK( subscriber.subscribe( *largeEchoEvent ));
+    BOOST_CHECK( subscriber.subscribe( test::Echo::IDENTIFIER(),
```
Comment (hernando) [inline tests/pubSub.cpp]: To avoid confusion, move the declaration of message closer to this statement.

File: tests/connection/broker.cpp (hunk 1 above)
Comment (tribal-tec) [inline tests/connection/broker.cpp]: Nope, because of https://github.com/HBPVIS/ZeroEQ/pull/168

File: tests/receiver.cpp (hunk 2 above)
Comment (tribal-tec) [inline tests/receiver.cpp]: Won't save anything here, plus it's just a unit test :)

**Comment by tribal-tec** [comment]
ping

**Comment by eile** [comment]
+1 from me

**Pull request merged**